src/stats.c

```diff
@@ -86,11 +86,16 @@ void _mi_stat_adjust_decrease(mi_stat_count_t* stat, size_t amount) {
 static void mi_stat_count_add_mt(mi_stat_count_t* stat, const mi_stat_count_t* src) {
   if (stat==src) return;
   mi_atomic_void_addi64_relaxed(&stat->total, &src->total); 
-  mi_atomic_void_addi64_relaxed(&stat->current, &src->current); 
+  
+  int64_t prev_current = mi_atomic_addi64_relaxed(&stat->current, src->current);
+
   // peak scores do really not work across threads .. we just add them
-  mi_atomic_void_addi64_relaxed( &stat->peak, &src->peak);
+  //mi_atomic_void_addi64_relaxed( &stat->peak, &src->peak);
```

Comment:
Unfortunately multi-line comments are very buggy on GitHub.

I just wanted to point out to move the comment where it belongs (above the new mi_atomic_addi64_relaxed) call, rephrase it and remove the commented out line.

```suggestion
  // peak scores don't work well across threads, approximate them
  int64_t prev_current = mi_atomic_addi64_relaxed(&stat->current, src->current);
```